ztp/gitops-subscriptions/argocd/SNOExpansion.md

```diff
@@ -0,0 +1,483 @@
+  namespace: $SNO_NAME
+spec:
+  scope:
+    resource: sriovoperatorconfigs
+    name: default
+    namespace: openshift-sriov-network-operator
+EOF
+```
+Check the operator config:
+
+```bash
+$ oc get managedclusterview check-sriov-node-selector -n $SNO_NAME -ojsonpath='{ .status.result.spec }'
+{"disableDrain":true,"enableInjector":true,"enableOperatorWebhook":true,"logLevel":2}
+```
+If there is `configDaemonNodeSelector`, and it is set to `node-role.kubernetes.io/master: ""`, it must be cleared.
+
+To clear the SR-IOV operator `configDaemonNodeSelector`, deploy the following manifest on your hub cluster:
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: action.open-cluster-management.io/v1beta1
+kind: ManagedClusterAction
+metadata:
+  name: clear-sriov-node-selector
+  namespace: $SNO_NAME
+spec:                 
+  actionType: Delete
+  kube:             
+    resource: sriovoperatorconfigs
+    name: default
+    namespace: openshift-sriov-network-operator
+EOF              
+```
+
+Monitor the outcome (within the next minute) by:
+```bash
+oc get managedclusteraction/clear-sriov-node-selector -n $SNO_NAME -o jsonpath='{.status.conditions[*]}'
+{"lastTransitionTime":"2022-08-21T08:39:04Z","message":"Resource action is done.","reason":"ActionDone","status":"True","type":"Completed"}
+```
+
+Delete the managedclusterview to free up resources:
+
+```bash
+$ oc delete managedclusterview check-sriov-node-selector -n $SNO_NAME
+```
+##### __Redeploy SNO policies__
+Redeploy the common policies using ztp plugin container version 4.12 or above using TALM operator custom resource.
+
+#### __Preparing worker node policies__
+Create the following policy template:
+
+```yaml
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "cnfdf15-worker"
+  namespace: "ztp-cnfdf15-policies"
+spec:
+  bindingRules:
+    # These policies will correspond to all clusters with this label:
+    sites: "cnfdf15"
+######
+# Comment 1
+  mcp: "worker"
+  sourceFiles:
+    - fileName: PtpConfigSlave.yaml   # Change to PtpConfigSlaveCvl.yaml for ColumbiaVille NIC
+      policyName: "config-policy"
+      metadata:
+        name: "du-ptp-slave-worker"
+      spec:
+        profile:
+        - name: "slave-worker"
+          # This interface must match the hardware in this group
+######
+# Comment 2
+          interface: "ens1f0"
+          ptp4lOpts: "-2 -s --summary_interval -4"
+          phc2sysOpts: "-a -r -n 24"
+        recommend:
+        - match:
+          - nodeLabel: kubernetes.io/hostname=cnfdf16
+          priority: 4
+          profile: slave-worker
+    - fileName: SriovNetwork.yaml
+######
+# Comment 3    
+      policyName: "config-policy-net"
+      metadata:
+        name: "f1u-network-worker"
+      spec:
+        resourceName: uplane_worker
+        vlan: 140
+    - fileName: SriovNetworkNodePolicy.yaml
+      policyName: "config-policy-net"
+      metadata:
+        name: "sriov-nnp-uplane-worker"
```

Comment:
Since this use case assumes workers are added at a later point in time, we have already deployed DU profile to the SNO with all the selectors set to "master". We are not going to reconfigure the master, as this will incur downtime (and we are required to avoid downtime). So we can only talk about equality of nodes between the additional workers. These could have been sharing a common configuration, but we don't have any label applied to workers that makes them unique (because worker label is applied to the original SNO too).
I think the right solution would be to implement node labeling API as a part of siteconfig. This will help us in other places as well (such as ODF storage nodes in CU deployments for example). It will also allow us to stop using master and worker labels everywhere as we do now.